.github/workflows/pr-line-check.yml

```diff
@@ -0,0 +1,64 @@
+name: Check PR Lines Changed
+
+on:
+  workflow_call:
+    inputs:
+      max_lines:
+        description: 'Maximum allowed total lines changed'
+        required: false
+        type: number
+        default: 1000
+      base_ref:
+        description: 'Default base branch to compare against (if not running on a PR)'
+        required: false
+        type: string
+        default: 'main'
+      ignore_patterns:
+        description: 'Regex pattern for files to ignore when calculating changes'
+        required: false
+        type: string
+        default: '(\.lock$)'
+
+jobs:
+  check-lines:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Determine base branch
+        id: get-base-branch
+        run: |
+          # Use the PR base branch if available; otherwise use the default input.
+          if [ -n "${{ github.event.pull_request.base.ref }}" ]; then
+            echo "Using PR base branch: ${{ github.event.pull_request.base.ref }}"
+            echo "base_branch=${{ github.event.pull_request.base.ref }}" >> "$GITHUB_OUTPUT"
+          else
+            echo "Using default base branch: ${{ inputs.base_ref }}"
+            echo "base_branch=${{ inputs.base_ref }}" >> "$GITHUB_OUTPUT"
+          fi
+
+      - name: Calculate changed lines
+        id: line_count
+        run: |
+          BASE_BRANCH="${{ steps.get-base-branch.outputs.base_branch }}"
+          # Fetch the base branch for comparison
+          git fetch origin "$BASE_BRANCH"
```

Comment:
mmm so we are fetching the entire repo here right? In Extension, we avoided doing a full fetch, and instead we perform a fetch with depth, to optimize the flow [here](https://github.com/MetaMask/metamask-extension/blob/main/.circleci/config.yml#L69) and we also have [this function which increments the depth](https://github.com/MetaMask/metamask-extension/blob/main/.circleci/scripts/git-diff-default-branch.ts#L49), in case it's needed (for doing the git-diff operation later on)

I'm wondering if we could re-use some of this? Otherwise we are performing a git fetch with depth, but with this flow, were are now also performing a full fetch :thinking:

Reply:
So would the change here be to update row 46 to `git fetch --depth 1 origin "$BASE_BRANCH"`? I don't think I need the function referenced but please correct me if I am missing something.

Reply:
I think you would need to fetch a depth of more than 1, as you need all commits in-between the base branch and the HEAD to be present. I cannot recall the behavior of `git diff` when the commits needed for the diff are missing, but surely it will either error or return suboptimal results without the commits present that are needed for the diff. I think it errors.

In the function @seaona linked, we increment the depth until the command `git merge-base origin/HEAD HEAD` succeeds, as the goal is to at least have commits up to the merge-base present. I believe a similar strategy should work here as well. Except that instead of `origin/HEAD` we may want to use `origin/$BASE_BRANCH`.

Reply:
I will say though that fetching the whole repository doesn't take _that_ long. Personally I'd consider this a non-blocking suggestion/optimization, which we could do later.

Reply:
@Gudahtt and @seaona I believe I addressed the feedback.